json_spec/output/test_step_artifact.json

```diff
@@ -0,0 +1,190 @@
+                }
+            },
+            "required": [
+                "testStepEnd"
+            ]
+        },
+        {
+            "properties": {
+                "measurement": {
+                    "description": "TODO: link to md",
+                    "$ref": "/opencomputeproject/ocp-diag-core/measurement"
+                }
+            },
+            "required": [
+                "measurement"
+            ]
+        },
+        {
+            "properties": {
+                "measurementSeriesStart": {
+                    "description": "TODO: link to md",
+                    "$ref": "/opencomputeproject/ocp-diag-core/measurementSeriesStart"
+                }
+            },
+            "required": [
+                "measurementSeriesStart"
+            ]
+        },
+        {
+            "properties": {
+                "measurementSeriesEnd": {
+                    "description": "TODO: link to md",
+                    "$ref": "/opencomputeproject/ocp-diag-core/measurementSeriesEnd"
+                }
+            },
+            "required": [
+                "measurementSeriesEnd"
+            ]
+        },
+        {
+            "properties": {
+                "measurementSeriesElement": {
+                    "description": "TODO: link to md",
+                    "$ref": "/opencomputeproject/ocp-diag-core/measurementSeriesElement"
+                }
+            },
+            "required": [
+                "measurementSeriesElement"
+            ]
+        },
+        {
+            "properties": {
+                "error": {
+                    "description": "TODO: link to md",
+                    "$ref": "/opencomputeproject/ocp-diag-core/error"
+                }
+            },
+            "required": [
+                "error"
+            ]
+        },
+        {
+            "properties": {
+                "log": {
+                    "description": "TODO: link to md",
+                    "$ref": "/opencomputeproject/ocp-diag-core/log"
+                }
+            },
+            "required": [
+                "log"
+            ]
+        },
+        {
+            "properties": {
+                "diagnosis": {
+                    "description": "TODO: link to md",
+                    "$ref": "/opencomputeproject/ocp-diag-core/diagnosis"
+                }
+            },
+            "required": [
+                "diagnosis"
+            ]
+        },
+        {
+            "properties": {
+                "file": {
+                    "$ref": "#/$defs/file"
+                }
+            },
+            "required": [
+                "file"
+            ]
+        },
+        {
+            "properties": {
+                "extension": {
+                    "$ref": "#/$defs/extension"
+                }
+            },
+            "required": [
+                "extension"
+            ]
+        }
+    ],
+    "$defs": {
+        "file": {
```

Comment:
Should we pull this out into it's own file (and others like it) to better match the spec doc even if it isn't referenced elsewhere?

Reply:
ive considered that, but generally stuck to the rule of "referenced in more than 1 place". let me see how it looks as a different file and i'll update.

Reply:
done, please close after check